### PR TITLE
Add ability to read in a dictionary in the repl

### PR DESCRIPTION
**Pull request by HuwCampbell**

### CODE REVIEW

File: src/Icicle/Dictionary/Parse.hs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      structField = do
+        n <- takeWhile (/= ':')
+        _ <- char ':'
+        e <- encoding
+        o <- Optional <$ char '*' <|> pure Mandatory
+        pure $ StructField o (Attribute n) e
+
+parseDictionaryLineV1 :: Text -> Either ParseError DictionaryEntry
+parseDictionaryLineV1 s =
+  eitherify $ parse parseIcicleDictionaryV1 s
```
Comment (tranma) [inline src/Icicle/Dictionary/Parse.hs]: `parseOnly`?

File: main/icicle-repl.hs [hunk 2]
```diff
@@ -189,12 +192,20 @@ handleLine state line = case readCommand line of
...
       Right fs -> do
         HL.outputStrLn $ "ok, loaded " <> fp <> ", " <> show (length fs) <> " rows"
         return $ state { facts = fs }
 
+  Just (CommandLoadDictionary fp) -> do
+    s  <- liftIO $ T.readFile fp
+    case SR.readDictionary s of
+      Left e   -> prettyHL e >> return state
+      Right ds -> do
+        HL.outputStrLn $ "ok, loaded " <> fp
```
Comment (amosr) [inline main/icicle-repl.hs]: it's silly but maybe just something like "loaded dictionary with 5 fields".
if CommandSetShow showed the dictionary too it'd be super neato

File: src/Icicle/Dictionary/Parse.hs (hunk 1 above)
Comment (HuwCampbell) [inline src/Icicle/Dictionary/Parse.hs]: Yep

**Comment by amosr** [comment]
test? test/cli/repl/*
:monkey_face:

**Comment by amosr** [comment]
:no_mobile_phones:

**Pull request merged**